joomla/joomla-cms
# [4.0] Mail Templates improvement

Pull Request for Issue #27619.

### Summary of Changes
This PR makes some improvements to Mail Templates to make it easier to use (I hope :D ). Below are the changes (base on suggestions from https://github.com/joomla/joomla-cms/issues/27619)

1. Link Title column to edit mail template of the default language
2. Remove **Templates** and No **Templates** columns
3. On a site with multiple content languages, add **Edit Templates** column to allow click on to edit mail template of a specific language
4. Remove switcher, show actual mail template content instead of just language item
5. Add **Reset To Default** to allow reset to default subject, body and html body of a mail template
6. Replace **Available Tags** with **Insert Tags**
7. Show Default Content For HTML Body (base on body with nl2br method applied) if there is no data for HTML Body from master template

Below are some screenshot to show the changes:

- Mail Templates on a single/mono language
![mono-language](https://user-images.githubusercontent.com/977664/115690204-02ee0d80-a387-11eb-9311-478942b7abd1.png)

- Mail Templates on multiple languages
![multiple_languages](https://user-images.githubusercontent.com/977664/115690293-1d27eb80-a387-11eb-8869-cbcdc160e921.png)

- Edit Mail Template of a language
![edit_mail_template](https://user-images.githubusercontent.com/977664/115690301-2022dc00-a387-11eb-91b9-57fec6a2f46f.png)

### Testing Instructions
Download update package for this PR at https://ci.joomla.org/artifacts/joomla/joomla-cms/4.0-dev/33230/downloads/42268/Joomla_4.0.0-beta8-dev+pr.33230-Development-Update_Package.zip, go to System -> Update -> Joomla to update. Then try to use Mail Templates to see if it is working OK. Hopefully it is easier to use than the default implementation.
